// wslink/webpack.config.js
var path = require('path');

module.exports = {
  context: path.resolve(__dirname),
  entry: ['./src/index.js'],
  output: {
    filename: 'wslink.js',
    path: path.resolve(__dirname, 'dist'),
    library: "wslink",
    libraryTarget: "umd"
  },
  mode: 'development',
  devtool: 'source-map',
  module: {
    rules: [
      {
        test: /\.js$/,
        use: [
          {
            loader: 'babel-loader',
            options: {
              presets: ['@babel/preset-env'],
            },
          },
		  {
            loader: 'style-loader'
          },
          {
            loader: 'css-loader'
          },
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: () => [
                  require('autoprefixer')
                ]
              }
            }
          },
          {
            loader: 'sass-loader'
          }
        ],
      },
    ]
  },
  
devServer: {
    allowedHosts: 'auto' | 'all' | Array[string]
}

};
